lib/utils/hasRulesUsingLexer.mjs

```diff
@@ -0,0 +1,19 @@
+import getStylelintRule from './getStylelintRule.mjs';
+
+/**
+ * @param {import('stylelint').Config} config
+ * @returns {Promise<boolean>}
+ */
+export default async function hasRulesUsingLexer(config) {
+	const ruleNames = config.rules ? Object.keys(config.rules) : [];
```

Comment:
Maybe return `false` if `rules?.length` is falsy?

Reply:
Fixed in this commit!
[Refactor check rule length and early return](https://github.com/stylelint/stylelint/pull/8297/commits/6997f04434228e05fb129fc0e9bf7844cf3cf52f)